mpcs/src/util.rs

```diff
@@ -148,38 +148,45 @@ pub fn field_type_index_set_ext<E: ExtensionField>(
     }
 }
 
-pub struct FieldTypeIterExt<'a, E: ExtensionField> {
-    inner: &'a FieldType<E>,
-    index: usize,
+pub fn poly_iter_ext<E: ExtensionField>(
+    poly: &DenseMultilinearExtension<E>,
+) -> impl Iterator<Item = E> + '_ {
+    field_type_iter_ext(&poly.evaluations)
 }
 
-impl<'a, E: ExtensionField> Iterator for FieldTypeIterExt<'a, E> {
-    type Item = E;
+pub fn field_type_iter_ext<E: ExtensionField>(
+    evaluations: &FieldType<E>,
+) -> impl Iterator<Item = E> + '_ {
+    match evaluations {
+        FieldType::Ext(coeffs) => Either::Left(coeffs.iter().copied()),
+        FieldType::Base(coeffs) => Either::Right(coeffs.iter().map(|x| (*x).into())),
+        _ => unreachable!(),
+    }
+}
 
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.index >= self.inner.len() {
-            None
-        } else {
-            let res = field_type_index_ext(self.inner, self.index);
-            self.index += 1;
-            Some(res)
-        }
+pub fn field_type_to_ext_vec<E: ExtensionField>(evaluations: &FieldType<E>) -> Vec<E> {
+    match evaluations {
+        FieldType::Ext(coeffs) => coeffs.to_vec(),
+        FieldType::Base(coeffs) => coeffs.iter().map(|x| (*x).into()).collect(),
```

Comment:
Cosmetic suggestion:

```suggestion
        FieldType::Base(coeffs) => coeffs.iter().map(|&x| x.into()).collect(),
```